app/server_inactivity.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+const serverInactivityHours = 100
+
+func (s *Server) doInactivityCheck() {
+	if !s.Config().FeatureFlags.EnableInactivityCheckJob {
+		mlog.Info("No activity check because EnableInactivityCheckJob feature flag is disabled")
+		return
+	}
+
+	inactivityDurationHourseEnv := os.Getenv("MM_INACTIVITY_DURATION")
+	inactivityDurationHours, parseError := strconv.ParseFloat(inactivityDurationHourseEnv, 64)
+	if parseError != nil {
+		// default to 100 hours
+		inactivityDurationHours = serverInactivityHours
+	}
+
+	systemValue, sysValErr := s.Store.System().GetByName("INACTIVITY")
+	if sysValErr != nil {
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Warn("An error occurred while getting INACTIVITY from system store", mlog.Err(sysValErr))
+		}
```

Comment:
Is this the case if there is no store value for `INACTIVITY`?

Reply:
I am going to consider logging for any error encountered regardless of whether it is a not found. I think it's safe since we have a nil check down for this `systemValue`

```go
systemValue, sysValErr := s.Store.System().GetByName("INACTIVITY")
	if sysValErr != nil {
		mlog.Warn("An error occurred while getting INACTIVITY from system store", mlog.Err(sysValErr))
	}
```

Reply:
Thanks for clarifying!